de-lib/src/test/java/org/iplantc/de/apps/client/presenter/toolBar/AppsToolbarPresenterImplTest.java

```diff
@@ -193,4 +210,46 @@
                                appServiceMock);
     }
 
+    @Test
+    public void verifyServiceCalled_onPublishSelected_Publishable() {
+        PublishAppSelected eventMock = mock(PublishAppSelected.class);
+        SubmitAppForPublicDialog dlgMock = mock(SubmitAppForPublicDialog.class);
+        App app = mock(App.class);
+        when(app.getSystemId()).thenReturn("DE");
+        when(app.getId()).thenReturn("1");
+        when(eventMock.getApp()).thenReturn(app);
+
+        uut.onPublishAppSelected(eventMock);
+
+        verify(appServiceMock).isPublishable(eq("DE"), eq("1"), booleanCallbackCaptor.capture());
+
+        booleanCallbackCaptor.getValue().onSuccess(true);
+        verify(submitAppDialogAsyncProviderMock).get(submitDialogCaptor.capture());
+        submitDialogCaptor.getValue().onSuccess(dlgMock);
+        verify(dlgMock).show(eq(app));
+    }
+
+    @Test
+    @Ignore
+    //TODO: Fix this test
+    /**
+     * This test will fail currently because AlertMessageBox initalization fails. Have to dig up mocking library.
```

Comment:
You could probably skip that issue by putting the creation of that `AlertMessageBox` in a method that you override during testing.  Or really, I don't think we absolutely need to test the `false` condition since it just displays a warning.  And the `true` condition is tested above, which I think is sufficient.